src/shared/claim-page.tsx

```diff
@@ -4,24 +4,31 @@ import {
   Grid,
   Heading,
   Icon,
-  Image,
   Skeleton,
   Text,
 } from "@chakra-ui/react";
-import { DropContract, useContractMetadata } from "@thirdweb-dev/react";
+import {
+  DropContract,
+  ThirdwebNftMedia,
+  useContractMetadata,
+  useNFT,
+} from "@thirdweb-dev/react";
 import React from "react";
 import { DropSvg } from "./svg/drop";
 
 interface ClaimPageProps {
   contract?: Exclude<DropContract, null>;
+  tokenId?: string;
 }
 
 export const ClaimPage: React.FC<React.PropsWithChildren<ClaimPageProps>> = ({
   contract,
-
+  tokenId,
   children,
 }) => {
   const { data: metadata, isLoading } = useContractMetadata(contract);
+  const { data: nft } = useNFT(contract, tokenId);
+
```

Comment:
I would like to split this into 2 again now, `TokenClaimPage` (with the token) and `ContractClaimPage` (without the token)

we shouldn't even hav useNFT in the contract claim page case and we might not need contract metadata in the token claim page case? might just want to render token name and description instead?